// usersw/build/components/SelectUserModal/UserList/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _antdMobile = require("antd-mobile");

require("antd-mobile/lib/icon/style/css.js");

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var UserList =
/*#__PURE__*/
function (_Component) {
  _inherits(UserList, _Component);

  function UserList() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, UserList);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(UserList)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "handleSelected", function (userInfo) {
      var onSelected = _this.props.onSelected;

      if (onSelected) {
        onSelected(userInfo);
      }
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "renderIdentity", function () {
      var userType = _this.props.userType;
      return _react.default.createElement("div", {
        className: "select-user-list-title"
      }, userType === 'student' ? '学生' : '教职工');
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "renderUserItem", function (_ref) {
      var userInfo = _ref.userInfo,
          isLast = _ref.isLast;
      var _this$props = _this.props,
          currentUserInfo = _this$props.currentUserInfo,
          replaceUserName = _this$props.replaceUserName;
      var orgId = userInfo.orgId,
          orgName = userInfo.orgName,
          _userInfo$userId = userInfo.userId,
          userId = _userInfo$userId === void 0 ? '' : _userInfo$userId,
          thisUserName = userInfo.userName;
      var cls = (0, _classnames.default)(['select-user-item', !isLast && 'line-bottom']);
      var isSelected = currentUserInfo.userId === userId;
      var userName = replaceUserName ? null : thisUserName;
      return _react.default.createElement("div", {
        className: cls,
        key: "".concat(orgId, "_").concat(userId),
        onClick: function onClick() {
          return _this.handleSelected(userInfo);
        }
      }, _react.default.createElement("div", {
        className: "select-user-item-user-name"
      }, userName), _react.default.createElement("div", {
        className: "select-user-item-org-name ellipsis"
      }, orgName), isSelected && _react.default.createElement(_antdMobile.Icon, {
        type: "check"
      }));
    });

    return _this;
  }

  _createClass(UserList, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props2 = this.props,
          _this$props2$userList = _this$props2.userList,
          userList = _this$props2$userList === void 0 ? [] : _this$props2$userList,
          userType = _this$props2.userType;
      var thisUserList = userList && userList.filter(function (user) {
        return user.userType === userType;
      }) || [];
      console.log(thisUserList, 'thisUserList');

      if (thisUserList && thisUserList.length === 0) {
        return null;
      }

      return _react.default.createElement("div", {
        className: "select-user-list"
      }, this.renderIdentity(), _react.default.createElement("div", {
        className: "select-user-list-body line-bottom"
      }, thisUserList.map(function (userInfo, index) {
        return _this2.renderUserItem({
          userInfo: userInfo,
          isLast: thisUserList.length - 1 === index
        });
      })));
    }
  }]);

  return UserList;
}(_react.Component);

var _default = UserList;
exports.default = _default;
