// usersw/build/service/service-utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setEnv = setEnv;
exports.url = exports.INDUSTRYTYPE = exports.APPNAME = exports.APPID = exports.CORPID = void 0;

var _utils = require("./utils.js");

/* eslint-disable import/no-mutable-exports */
var APIEnv = _utils.ENV;
var CORPID;
exports.CORPID = CORPID;
var APPID;
exports.APPID = APPID;
var APPNAME;
exports.APPNAME = APPNAME;
var INDUSTRYTYPE;
exports.INDUSTRYTYPE = INDUSTRYTYPE;

function setEnv(params) {
  var _params$APIEnv = params.APIEnv,
      newENV = _params$APIEnv === void 0 ? 'dev' : _params$APIEnv,
      _params$corpid = params.corpid,
      corpid = _params$corpid === void 0 ? '' : _params$corpid,
      _params$appId = params.appId,
      appId = _params$appId === void 0 ? '' : _params$appId,
      _params$appName = params.appName,
      appName = _params$appName === void 0 ? '' : _params$appName,
      _params$industryType = params.industryType,
      industryType = _params$industryType === void 0 ? '' : _params$industryType;
  APIEnv = newENV;
  exports.CORPID = CORPID = corpid;
  exports.APPID = APPID = appId;
  exports.APPNAME = APPNAME = appName;
  exports.INDUSTRYTYPE = INDUSTRYTYPE = industryType;
}

var urlCollect = {
  // 开发环境
  dev: {
    url: '//gateway.community-dev.easyj.top/auth',
    usercenter: '//gateway.community-dev.easyj.top/user-center'
  },
  // 测试环境
  sit: {
    url: '//gateway.community-sit.easyj.top/auth',
    usercenter: '//gateway.community-sit.easyj.top/user-center'
  },
  // 生产环境
  production: {
    url: '//gateway.suosi.atuniversity.cn/auth',
    usercenter: '//gateway.community-dev.easyj.top/user-center'
  }
};
var url = urlCollect[APIEnv];
exports.url = url;
